fix: release workflow fails with 'tar: file changed as we read it'

## Bug

The release GitHub Action fails on the **Build backend bundle** step with exit code 1:

\`\`\`
tar: .: file changed as we read it
##[error]Process completed with exit code 1.
\`\`\`

**Run:** https://github.com/codemonkeybr/meshcore-pi-companion/actions/runs/24754970408

## Root Cause

\`tar\` writes the output \`.tar.gz\` file into the same directory (\`.\`) it is archiving. Even though \`--exclude="./$artifact"\` is set, \`tar\` still detects the directory changed while it was being read (because the output file is created before the scan completes), and exits with code 1.

## Fix

Write the tarball to \`/tmp/$artifact\` first, then \`mv\` it back to the working directory. This eliminates the race condition entirely and removes the need for the self-exclusion flag.

\`\`\`diff
-          tar czf "$artifact" \
+          tar czf "/tmp/$artifact" \
             ...
-            --exclude="./$artifact" \
             .
+          mv "/tmp/$artifact" .
\`\`\`

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -54,7 +54,7 @@ jobs:
           set -euo pipefail
           version="${{ steps.version.outputs.version }}"
           artifact="remoteterm-backend-${version}.tar.gz"
-          tar czf "$artifact" \
+          tar czf "/tmp/$artifact" \
             --exclude='./.git' \
             --exclude='./.github' \
             --exclude='./.venv' \
@@ -71,8 +71,8 @@ jobs:
             --exclude='./README.md' \
             --exclude='./CLAUDE.md' \
             --exclude='./AGENTS.md' \
-            --exclude="./$artifact" \
             .
+          mv "/tmp/$artifact" .
           sha256sum "$artifact" > "${artifact}.sha256"
           ls -lh "$artifact" "${artifact}.sha256"
           # Sanity check: bundle must contain scripts/ and pyproject.toml.

